Python_Scripts/plot_NonLinSol_Supp.py

```diff
@@ -20,198 +21,209 @@
-    for iTick in [2, 5, 8, 11, 14, 17, 20]:
-        specific_xticks_minor[iTick].set_visible(True)
-
-    # plot text 'B'
-    ax2.text(-0.13, 1, 'B', fontsize=labelsize + 5, transform=ax2.transAxes)
-
-
-########## call both functions + some global properties
-NonLinearSolver('2','9')
+solAlg = '2'
+LinSol = '9'
+
+###############################################################################
+# subplot 1: linear regressions of combined scatter plots
+
+# list of all 14 data frames for better indexing in the future
+all_intern_columns = [pd.DataFrame(columns=[]) for _ in range(14)]
+column_names = []
+
+base_path = DIR_DATA_WHOLESTUDY
+
+# choose only the correct files
+all_files = sorted(os.listdir(base_path))
+correct_files = []
+for iFile in range(0, len(all_files)):
+    if all_files[iFile].split('_')[0] == solAlg and all_files[iFile].split('_')[2] == LinSol:
+        correct_files.append(all_files[iFile])
+
+# open all .tsv linear solver files + save right column in data frame
+for iCorrectFile in range(0, len(correct_files)):  # each .tsv file
+    next_tsv = pd.read_csv(os.path.join(
+        base_path, correct_files[iCorrectFile]), sep='\t')
+
+    # reset after each iteration
+    next_time_value = []
+    num_x = []
+
+    # open next file
+    next_tsv = averaging(next_tsv)
+
+    # get the correct values
+    for iFile in range(0, len(next_tsv['id'])):
+        if next_tsv['t_intern_ms'][iFile] != 0:
+            next_time_value.append(next_tsv['t_intern_ms'][iFile])
+            num_x.append(next_tsv['state_variables'][iFile])
+
+    # append new column to existing data frame with correct log10 values
+    column_names.append('simulation_time')
+    all_intern_columns[iCorrectFile]['state_variables'] = pd.Series(num_x)
+    all_intern_columns[iCorrectFile]['simulation_time'] = \
+        pd.Series(next_time_value)
+
+# plot scatter plot and linear regressions of all data points for the
+#  accompanying linear solver
+fontsize = 12
+labelsize = 8
+alpha = 0.5
+marker_size = 2
+ax1.set_xscale('log')
+ax1.set_yscale('log')
+ax1.set_xlim([0.8, 2000])
+ax1.set_ylim([0.2, 100000])
+ax1.set_xlabel('Number of state variables', fontsize=fontsize)
+ax1.set_ylabel('Simulation time [ms]', fontsize=fontsize)
+ax1.tick_params(labelsize=labelsize)
+
+linSol_for_Legend = ['Functional', 'Newton-type']
+colors = ['#e66101', '#5e3c99']
+for iLinearSolverDataPoints in range(0, int(len(correct_files)/7)):
+    # concatenate Data Frames in categories of linear solver data points
+    vertically_stacked_tsv = pd.concat(
+        [all_intern_columns[7*iLinearSolverDataPoints],
+         all_intern_columns[7*iLinearSolverDataPoints + 1],
+         all_intern_columns[7*iLinearSolverDataPoints + 2],
+         all_intern_columns[7*iLinearSolverDataPoints + 3],
+         all_intern_columns[7*iLinearSolverDataPoints + 4],
+         all_intern_columns[7*iLinearSolverDataPoints + 5],
+         all_intern_columns[7*iLinearSolverDataPoints + 6]], axis=0)
```

Comment:
This seems to be solvable using list comprehension....

Reply:
Agreed. Feel free.